overlord/snapstate/backend/copydata.go

```diff
@@ -272,6 +272,17 @@ func (b Backend) InitExposedSnapHome(snapName string, rev snap.Revision) (err er
 		}
 
 		newUserHome := snap.UserExposedHomeDir(usr.HomeDir, snapName)
+		if exists, isDir, err := osutil.DirExists(newUserHome); err != nil {
+			return err
+		} else if exists {
+			if !isDir {
+				return fmt.Errorf("cannot initialize new user HOME %q: already exists but is not a directory", newUserHome)
+			}
+
+			// we reverted from a core22 base before, so the new HOME already exists
```

Comment:
we need a TODO somewhere that we need to make the higher level doCopyData aware of this to decide what to do on undo?